extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/StepUpAuthenticationPolicy.java

```diff
@@ -0,0 +1,140 @@
+package io.quarkus.oidc.runtime;
+
+import static io.quarkus.oidc.AuthenticationContext.NO_MAX_AGE;
+import static io.quarkus.oidc.common.runtime.OidcConstants.ACR_VALUES;
+import static io.quarkus.oidc.common.runtime.OidcConstants.MAX_AGE;
+import static io.quarkus.vertx.http.runtime.security.HttpSecurityUtils.getAuthenticationFailureFromEvent;
+import static io.quarkus.vertx.http.runtime.security.HttpSecurityUtils.getRoutingContextAttribute;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.function.Consumer;
+
+import org.eclipse.microprofile.jwt.Claims;
+import org.jboss.logging.Logger;
+
+import io.quarkus.security.AuthenticationFailedException;
+import io.quarkus.security.identity.request.TokenAuthenticationRequest;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.RoutingContext;
+
+record StepUpAuthenticationPolicy(String[] expectedAcrValues, long maxAge) implements Consumer<TokenVerificationResult> {
+
+    private static volatile boolean enabled = false;
+
+    StepUpAuthenticationPolicy(String acrValues, String maxAge) {
+        this(acrValues.isEmpty() ? null : acrValues.split(","), Long.parseLong(maxAge));
+    }
+
+    private static final Logger LOG = Logger.getLogger(StepUpAuthenticationPolicy.class);
+    private static final String AUTHENTICATION_POLICY_KEY = "io.quarkus.oidc.runtime.step-up-auth";
+
+    @Override
+    public void accept(TokenVerificationResult t) {
+        JsonObject json = t.localVerificationResult != null ? t.localVerificationResult
+                : new JsonObject(t.introspectionResult.getIntrospectionString());
+        if (expectedAcrValues != null) {
+            verifyAcr(json);
+        }
+        if (maxAge != NO_MAX_AGE) {
+            verifyMaxAge(json);
+        }
+    }
+
+    private void verifyMaxAge(JsonObject json) {
+        Long authTime = json.getLong(Claims.auth_time.name());
+        if (authTime != null) {
+            final long nowSecs = System.currentTimeMillis() / 1000;
+            final long expiresAtSecs = authTime + maxAge;
+            if (expiresAtSecs <= nowSecs) {
+                throwAuthenticationFailedException("Token issued at " + authTime + " is expired, because max age was set to "
```

Comment:
`auth_time` is not the same as the token `issued at` time, and the same for the expiry. Perhaps we can simplify and just say `the token age has exceeded %s`...